### PR TITLE
FEATURE: Replace share-popup with share-topic

### PR DESCRIPTION
**Pull request by nbianca**
share-topic modal is used everywhere except when clicking on the top
right corner of the post. This change standardizes on share-topic modal
and adds the missing features from share-popup.

Before:

image

After:

image

### CODE REVIEW

File: app/assets/javascripts/discourse/app/widgets/post.js [hunk 1]
```diff
@@ -280,21 +280,6 @@ createWidget("post-meta-data", {
...
-    const date = lastWikiEdit ? dateNode(lastWikiEdit) : dateNode(createdAt);
-    const attributes = {
-      class: "post-date",
-      href: attrs.shareUrl,
-      "data-share-url": attrs.shareUrl,
-      "data-post-number": attrs.post_number,
-    };
-
-    if (lastWikiEdit) {
-      attributes["class"] += " last-wiki-edit";
```
Comment (pmusaraj) [inline app/assets/javascripts/discourse/app/widgets/post.js:295]: We probably shouldn't remove the `last-wiki-edit` class here, it doesn't feel like it goes with the share-popup change?

File: app/assets/stylesheets/common/base/share_link.scss [hunk 2]
```diff
@@ -38,6 +38,13 @@
     display: flex;
     justify-content: flex-end;
     margin-left: auto;
+    align-items: baseline;
+
+    .date {
+      font-weight: normal;
+      color: var(--primary-med-or-secondary-med);
+      margin-right: 0.5em;
```
Comment (pmusaraj) [inline app/assets/stylesheets/common/base/share_link.scss:46]: Nitpick, but we only need the margin when the button is shown, so it's best to add a `.date + .btn { margin-left: 0.5em;}` rule instead.

**Review by pmusaraj — CHANGES_REQUESTED** [review]
Looks good, a couple of items (in addition to the code notes): 

- I see we use/extend `share-popup` in a customer theme and a plugin (discourse-docs). I believe we can use `share-topic` in those places before merging this. 
- The "+ New Topic" button should not be shown on the first post share. 
- I also think the "+ new topic" button is too prominent. Maybe we can use a simple link here with "Reply as linked topic" (and for messages "Reply as linked message") as the label?

File: app/assets/javascripts/discourse/app/widgets/post.js (hunk 1 above)
Comment (nbianca) [reply app/assets/javascripts/discourse/app/widgets/post.js:295]: This was moved to the new `post-date` widget below.

File: app/assets/stylesheets/common/base/share_link.scss (hunk 2 above)
Comment (nbianca) [reply app/assets/stylesheets/common/base/share_link.scss]: I moved the new topic button to the left, so the margin-right is no longer necessary. 👍🏻

**Comment by ZogStriP** [comment]
> The "+ New Topic" button should not be shown on the first post share.

Why not? It's the same behaviour as with the existing share-topic "modal"

image

> I also think the "+ new topic" button is too prominent. Maybe we can use a simple link here with "Reply as linked topic" (and for messages "Reply as linked message") as the label? Or maybe a flat button?

What's a flat button?

**Comment by pmusaraj** [comment]
> Why not? It's the same behaviour as with the existing share "modal"

True, I'm asking for a change to existing functionality. It feels odd to me to reply as a linked topic to an OP: why not reply just there, this is already a new topic? 

We can have this discussion separately, though, no need to hold up this PR for this. 

> What's a flat button?

It's the buttons that have no light gray background by default, usually used for small icon-only actions. Here's an example with an icon and a label: 

image

I like the "+ New Topic" button moved to the left, though. I'm happy with that.

**Review by pmusaraj — APPROVED** [review]

**Pull request merged**